File: all_sort/select_sort.py
#!/usr/bin/python3
# 选出最小的数放到序列首部
def select_sort(seq):
    n = len(seq)
    for i in range(n-1):
        min_idx = i
        for j in range(i+1, n):
            if seq[j] < seq[min_idx]:
                min_idx = j
        if i != min_idx:
            seq[min_idx], seq[i] = seq[i], seq[min_idx]
import random
lst = list(range(10))
random.shuffle(lst)
print(lst)
select_sort(lst)
print(lst)

File: all_sort/go/select_sort.go
package main

import "fmt"

func select_sort(seq []int) {
	l := len(seq)
	for i := 0; i < l-1; i++ {
		min_idx := i
		for j := i; j < l; j++ {
			if seq[j] < seq[min_idx] {
				min_idx = j
			}
		}
		if i != min_idx {
			seq[i], seq[min_idx] = seq[min_idx], seq[i]
		}
	}
}

func main() {
	var lst = []int{3, 1, 2, 5, 6, 7, 4}
	select_sort(lst)
	fmt.Printf("lst: %v\n", lst)
}

